src/utils/validators/index.ts

```diff
@@ -132,16 +132,27 @@ const validateRegexp = (regexp: string): boolean => {
   }
 };
 
+/**
+ * `validateNoSpecialCharacters` tests if a provided string contains no special characters
+ * @param str - The string to test.
+ * @returns - true if the string has no special characters and false otherwise
+ */
+const validateNoSpecialCharacters = (str: string): boolean => {
+  const noSpecialCharacters = /^[0-9a-zA-Z-._~()]*$/;
```

Comment:
It could be cool to define these special chars as a var that can be used in the error message so that any updates are reflected. I think this would work
```suggestion
const specialChars = "-._~()";
const validateNoSpecialCharacters = (str: string): boolean => {
    const noSpecialCharacters = new RegExp(`^[0-9a-zA-Z${specialChars}]*$`);
```